app/code/Magento/Review/Model/Review.php

```diff
@@ -6,32 +6,41 @@
 namespace Magento\Review\Model;
 
 use Magento\Catalog\Model\Product;
+use Magento\Framework\Api\AttributeValueFactory;
+use Magento\Framework\Api\ExtensionAttributesFactory;
+use Magento\Framework\Data\Collection\AbstractDb;
 use Magento\Framework\DataObject\IdentityInterface;
+use Magento\Framework\Model\AbstractModel;
+use Magento\Framework\Model\AbstractExtensibleModel;
+use Magento\Framework\Model\Context;
+use Magento\Framework\Model\ResourceModel\AbstractResource;
+use Magento\Framework\Registry;
+use Magento\Framework\UrlInterface;
+use Magento\Review\Api\Data\ReviewInterface;
 use Magento\Review\Model\ResourceModel\Review\Product\Collection as ProductCollection;
+use Magento\Review\Model\ResourceModel\Review\Product\CollectionFactory as ProductCollectionFactory;
 use Magento\Review\Model\ResourceModel\Review\Status\Collection as StatusCollection;
+use Magento\Review\Model\ResourceModel\Review\Status\CollectionFactory as StatusCollectionFactory;
+use Magento\Review\Model\ResourceModel\Review\Summary\CollectionFactory as SummaryCollectionFactory;
+use Magento\Review\Model\Review\Summary;
+use Magento\Review\Model\Review\SummaryFactory;
+use Magento\Store\Model\StoreManagerInterface;
 
 /**
  * Review model
  *
- * @api
- * @method string getCreatedAt()
- * @method \Magento\Review\Model\Review setCreatedAt(string $value)
- * @method \Magento\Review\Model\Review setEntityId(int $value)
- * @method int getEntityPkValue()
- * @method \Magento\Review\Model\Review setEntityPkValue(int $value)
- * @method int getStatusId()
- * @method \Magento\Review\Model\Review setStatusId(int $value)
+ * TODO: check if parent class can be changed to AbstractExtensibleModel; currently it is an AbstractModel
+ *
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
- * @since 100.0.2
  */
-class Review extends \Magento\Framework\Model\AbstractModel implements IdentityInterface
+class Review extends AbstractExtensibleModel implements IdentityInterface, ReviewInterface
```

Comment:
Lets try to avoid any changes in the existing models except for `AbstractModel` => `AbstractExtensibleModel` change and new interface.